app/models/link.rb

```diff
@@ -60,7 +58,12 @@ def link_with_matching_url
   end
 
   def existing_link_url
-    @_link ||= Link.where(url: self.url).distinct.first
+    link = Link.where(url: self.url).group_by(&:url)[self.url]
+    if link
+      @_link ||= link.first
+    else
+      @_link
+    end
```

Comment:
This had to be changed as a result of `link_errors` and `link_warnings` storing JSON.  After this change was made, the `distinct` methods stopped working due to a Postgres error `could not identify an equality operator for type json`.  After doing some research I found this could be fixed by changing the attributes' type to jsonb, however this requires a newer version of Postgres, so I used `group_by` as a workaround to avoid having to upgrade Postgres. Not very pretty, but it does the job!